packages/api/src/apps/api/app.ts

```diff
@@ -221,12 +221,16 @@ export default async (env: ProcessEnv) => {
   console.log("Started Express Server, API accessible");
 
   async function detectNewContracts(startBlock: number, endBlock: number) {
+    // ignore case if somehow start block and end block are out of whack
+    if (startBlock >= endBlock) return;
```

Comment:
When does this happen? Why does this happen?

Reply:
`startBlock == endBlock` happens only during testing where the loop runs several time before anything triggers a new block to be mined in the test.

Reply:
added an assert for that case and clarified comment to say this can happen if a block isnt advanced before loop runs again